nilearn/glm/first_level/first_level.py

```diff
@@ -1060,6 +1015,179 @@ def first_level_from_bids(dataset_path, task_label, space_label=None,
+    task_label : :obj:`str`
+        _description_
+    space_label : :obj:`str` or None
+        _description_
+    img_filters : list[Tuple[str, str]]
+        _description_
+    derivatives_path : :obj:`str`
+        _description_
+    supported_filters : list[str]
+        _description_
+
+    Raises
+    ------
+    TypeError
+        _description_
+    ValueError
+        _description_
+    """    
+    if not isinstance(dataset_path, str):
+        raise TypeError(
+            "dataset_path must be a string, "
+            f"instead {type(dataset_path)} was given"
+        )
+    if not os.path.exists(dataset_path):
+        raise ValueError(f'given path do not exist: {dataset_path}')
+
+    if not os.path.exists(derivatives_path):
+        raise ValueError('derivatives folder does not exist in given dataset')
+
+    if not isinstance(task_label, str):
+        raise TypeError(
+            "task_label must be a string, "
+            f"instead {type(task_label)} was given"
+        )
+
+    if space_label is not None and not isinstance(space_label, str):
+        raise TypeError(
+            "space_label must be a string, "
+            f"instead {type(space_label)} was given."
+        )
+
+    if not isinstance(img_filters, list):
+        raise TypeError(
+            f"img_filters must be a list, "
+            f"instead {type(img_filters)} was given."
+        )
+    for this_filter in img_filters:
+        if (not isinstance(this_filter[0], str)
+                or not isinstance(this_filter[1], str)):
+            raise TypeError('filters in img filters must be (str, str), '
+                            f"instead {type(this_filter)} was given.")
+        if this_filter[0] not in supported_filters:
+            raise ValueError(
+                f"field {this_filter[0]} is not a possible filter. "
+                f"Only {supported_filters} are allowed.")
+
+
+def _bids_filter(task_label: str,
+                 space_label: str | None = None,
+                 supported_filters: list[str] | None = None,
+                 extra_filter: list[Tuple[str, str]] | None = None
+                 ) -> list[Tuple[str, str]]:
+    """Return a filter to specific files from a BIDS dataset.
+
+    Parameters
+    ----------
+    task_label : :obj:`str`
+        _description_
+    space_label : :obj:`str` or None, optional
+        _description_
+    supported_filters : list[str] or None, optional
+        _description_
+    extra_filter : list[Tuple[str, str]] or None, optional
+        _description_
+
+    Returns
+    -------
+    list[Tuple[str, str]]
+        Filter to be used by :func:`get_bids_files`.
+    """    
+    filters = [('task', task_label)]
+    if space_label is not None:
+        filters.append(('space', space_label))
+    if extra_filter and supported_filters:
+        filters.extend(
+            this_filter
+            for this_filter in extra_filter
+            if this_filter[0] in supported_filters
+        )
```

Comment:
Maybe we should warn if other filters are given ?

Reply:
done